src/containers/StudentContainer/StudentContainer.js

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { makeStyles } from '@material-ui/core/styles';
+import { Link } from 'react-router-dom';
+import { Grid, Typography, Button } from '@material-ui/core';
+import { getStudentById } from '../../store/actions/studentsAction';
+
+const useStyles = makeStyles((theme) => ({
+  holder: {
+    margin: theme.spacing(4),
+  },
+  label: {
+    marginRight: theme.spacing(2),
+    fontWeight: 'bold',
+  },
+  title: {
+    marginBottom: theme.spacing(4),
+    fontWeight: 'bold',
+    fontSize: 25,
+  },
+  item: {
+    marginBottom: theme.spacing(2),
+  },
+  button: {
+    marginLeft: theme.spacing(2),
+  },
+}));
+
+const StudentContainer = (props) => {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+  const id = props.match.url.split('/')[3];
+  const studentById = useSelector((state) => state.students.student);
+
+  useEffect(() => {
+    dispatch(getStudentById(id));
+  }, [id]);
+
+  return (
+    <Grid container className={classes.holder} direction='column' justify='space-between'>
+      <Grid item>
+        <Typography className={classes.title}>Информация об ученике</Typography>
+      </Grid>
+      <Grid container className={classes.item} item>
+        <Typography className={classes.label}>Имя: </Typography>
+        <Typography>{studentById.firstName}</Typography>
+      </Grid>
```

Comment:
Повторяющийся блок вынести в отдельный компонент, а потом через деструктуризацию передавать данные.
```suggestion
const DataItem = ({gridClasses, labelClasses, data}) => (
      <Grid container className={gridClasses} item>
        <Typography className={labelClasses}>Имя: </Typography>
        <Typography>{data}</Typography>
      </Grid>
      );
      export default DataItem;
```

Reply:
@xx3u Осуществил рефакторинг, но без ДатаАйтема